File: mechanical/src/main/scala/resonantinduction/mechanical/process/grinder/GrinderNode.java
package resonantinduction.mechanical.process.grinder;

import net.minecraftforge.common.ForgeDirection;
import resonantinduction.core.interfaces.IMechanicalNode;
import resonantinduction.mechanical.energy.grid.MechanicalNode;

/** Node just for the grinder
 * 
 * @author Darkguardsman */
public class GrinderNode extends MechanicalNode
{
    public GrinderNode(TileGrindingWheel parent)
    {
        super(parent);
    }

    public TileGrindingWheel grider()
    {
        return (TileGrindingWheel) getParent();
    }

    @Override
    public boolean canConnect(ForgeDirection from, Object source)
    {
        if(grider().getDirection() == ForgeDirection.UP || grider().getDirection() == ForgeDirection.DOWN)
        {
            return grider().getDirection() == from || grider().getDirection().getOpposite() == from;
        }
        return grider().getDirection() != from && grider().getDirection().getOpposite() != from;
    }

    @Override
    public boolean inverseRotation(ForgeDirection dir, IMechanicalNode with)
    {
        return !(dir.offsetX > 0 || dir.offsetZ < 0 || dir.offsetY < 0);
    }
}

File: src/main/scala/resonantinduction/mechanical/mech/process/grinder/GrinderNode.scala
package resonantinduction.mechanical.mech.process.grinder

import net.minecraftforge.common.util.ForgeDirection
import resonantinduction.mechanical.mech.grid.MechanicalNode

/** Node just for the grinder
  *
  * @author Darkguardsman */
class GrinderNode(parent: TileGrindingWheel) extends MechanicalNode(parent: TileGrindingWheel)
{
  def grider: TileGrindingWheel =
  {
    return getParent.asInstanceOf[TileGrindingWheel]
  }

  override def canConnect[B](other: B, from: ForgeDirection): Boolean =
  {
    if (grider.getDirection == ForgeDirection.UP || grider.getDirection == ForgeDirection.DOWN)
    {
      return grider.getDirection == from || grider.getDirection.getOpposite == from
    }
    return grider.getDirection != from && grider.getDirection.getOpposite != from
  }


  override def inverseRotation(dir: ForgeDirection): Boolean =
  {
    return !(dir.offsetX > 0 || dir.offsetZ < 0 || dir.offsetY < 0)
  }
}